cli/config/addressbook.go

```diff
@@ -0,0 +1,158 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/addressbook"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+// AddressBook contains the configuration of the address book.
+type AddressBook struct {
+	// All is a map of all configured address entries in the address book.
+	All map[string]*AddressBookEntry `mapstructure:",remain"`
+}
+
+// Validate performs config validation.
+func (w *AddressBook) Validate() error {
+	// Make sure all entries are valid.
+	for name, acc := range w.All {
+		if err := config.ValidateIdentifier(name); err != nil {
+			return fmt.Errorf("malformed address name '%s': %w", name, err)
+		}
+
+		if err := acc.Validate(); err != nil {
+			return fmt.Errorf("address '%s': %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+// Load loads the given address.
+func (w *AddressBook) Load(name string) (*addressbook.Address, error) {
+	cfg, exists := w.All[name]
+	if !exists {
+		return nil, fmt.Errorf("address account '%s' does not exist in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return nil, fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	af := addressbook.AddressFactory{}
+	acc, err := af.Load(name, cfg.Config, cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure the address matches what we have in the config.
```

Comment:
This check is probably not needed here as the address is always just read from the config (I assume this is here because it was in the wallet where addresses are derived so may differ).